worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/PaperweightGetBlocks.java

```diff
@@ -670,12 +673,29 @@ public synchronized <T extends Future<T>> T call(IChunkSet set, Runnable finaliz
                                 if (entity != null) {
                                     final net.minecraft.nbt.CompoundTag tag = (net.minecraft.nbt.CompoundTag) adapter.fromNative(
                                             nativeTag);
+                                    if (entityTagMap.containsKey("Leash")) {
+                                        var leashTag = (net.minecraft.nbt.CompoundTag) adapter.fromNative(
+                                                entityTagMap.get("Leash")) ;
+                                        final LeashFenceKnotEntity leashEntity = leashRef.get(NbtUtils.readBlockPos(leashTag));
+                                        if (leashEntity != null) {
+                                            tag.put("Leash", NbtUtils.writeBlockPos(leashEntity.pos));
+                                        }
+                                    }
                                     for (final String name : Constants.NO_COPY_ENTITY_NBT_FIELDS) {
                                         tag.remove(name);
                                     }
                                     entity.load(tag);
                                     entity.absMoveTo(x, y, z, yaw, pitch);
                                     entity.setUUID(nativeTag.getUUID());
+                                    if (entity instanceof LeashFenceKnotEntity leashFenceKnotEntity) {
+                                        var leashTag = (net.minecraft.nbt.CompoundTag) adapter.fromNative(
+                                                entityTagMap.get("OldPos")) ;
+                                        if (leashTag != null) {
+                                            leashRef.put(new BlockPos(leashTag.getInt("X"), leashTag.getInt("Y"),
+                                                            leashTag.getInt("Z")),
```

Comment:
Unusual formatting, should it not either be a line per argument or all one line?

Reply:
One line per argument here 😄